utils_cv/common/plot.py

```diff
@@ -45,25 +45,38 @@ def line_graph(
 
 
 def show_ims(
-    im_paths: Union[str, List[str]],
+    im_or_im_paths: Union[str, List[str], np.ndarray, List[np.ndarray]],
     labels: Union[str, List[str]] = None,
     size: int = 3,
     rows: int = 1,
 ):
     """Show image files
     Args:
-        im_paths (str or List[str]): Image filepaths
+        im_or_im_paths (str or List[str] or numpy.ndarray or List[numpy.ndarray]): Image or image filepaths
         labels (str or List[str]): Image labels. If None, show image file name.
         size (int): MatplotLib plot size.
         rows (int): rows of the images
     """
-    if isinstance(im_paths, (str, Path)):
+    if im_or_im_paths is None or len(im_or_im_paths) == 0:
+        raise Exception(f"Empty {im_or_im_paths}")
+
+    # im_or_im_paths could be a numpy array of image filepaths
```

Comment:
Can this be simplified? I have been looking at this PR for 1.5 hours already, hence getting a bit tired. It's definitely not easy to read. I find it confusing that a single input variable name can either be (i) a single image, (ii) a list of images, (iii) a single image path or (iv) a list of image paths.

Reply:
Yeah, reverted.